hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/BatchStageImpl.java

```diff
@@ -165,6 +165,21 @@ public BatchStageImpl(@Nonnull Transform transform, FunctionAdapter ignored, @No
         return attachHashJoin(stage1, joinClause1, mapToOutputFn);
     }
 
+    @Nonnull @Override
+    public <K, T1_IN, T1, R> BatchStage<R> innerHashJoin(
+            @Nonnull BatchStage<T1_IN> stage1,
+            @Nonnull JoinClause<K, ? super T, ? super T1_IN, ? extends T1> joinClause1,
+            @Nonnull BiFunctionEx<T, T1, R> mapToOutputFn
+    ) {
+        BiFunctionEx<T, T1, R> finalOutputFn = (leftSide, rightSide) -> {
+            if (leftSide == null || rightSide == null) {
+                return null;
+            }
+            return mapToOutputFn.apply(leftSide, rightSide);
+        };
+        return attachHashJoin(stage1, joinClause1, finalOutputFn);
+    }
+
```

Comment:
We also have `hashJoin2()` and `hashJoinBuilder()`, they also need the same API extension. You could add a method to the hash join builder: `buildInner`.

Reply:
Maybe not builderInner, but rather addInner? So you can have left outer and inner join in one builder

Reply:
Yes, sounds better than my idea.